frontend/apps/reader/modules/readerhighlight.lua

```diff
@@ -916,42 +1010,57 @@ function ReaderHighlight:showHighlightNoteOrDialog(page, index, bookmark_note)
 end
 
 function ReaderHighlight:onShowHighlightDialog(page, index, is_auto_text)
-    local buttons = {
+    local row = {
         {
-            {
-                text = _("Delete"),
-                callback = function()
-                    self:deleteHighlight(page, index)
-                    UIManager:close(self.edit_highlight_dialog)
-                    self.edit_highlight_dialog = nil
-                end,
-            },
-            {
-                text = C_("Highlight", "Style"),
-                callback = function()
-                    self:editHighlightStyle(page, index)
-                    UIManager:close(self.edit_highlight_dialog)
-                    self.edit_highlight_dialog = nil
-                end,
-            },
-            {
-                text = is_auto_text and _("Add note") or _("Edit note"),
-                callback = function()
-                    self:editHighlight(page, index)
-                    UIManager:close(self.edit_highlight_dialog)
-                    self.edit_highlight_dialog = nil
-                end,
-            },
-            {
-                text = "…",
-                callback = function()
-                    self.selected_text = self.view.highlight.saved[page][index]
-                    self:onShowHighlightMenu(page, index)
-                    UIManager:close(self.edit_highlight_dialog)
-                    self.edit_highlight_dialog = nil
-                end,
-            },
-        }
+            text = _("Delete"),
+            callback = function()
+                self:deleteHighlight(page, index)
+                UIManager:close(self.edit_highlight_dialog)
+                self.edit_highlight_dialog = nil
+            end,
+        },
+        {
+            text = C_("Highlight", "Style"),
+            callback = function()
+                self:editHighlightStyle(page, index)
+                UIManager:close(self.edit_highlight_dialog)
+                self.edit_highlight_dialog = nil
+            end,
+        },
+    }
+    if Screen:isColorScreen() then
+        table.insert(row, {
+            text = C_("Highlight", "Color"),
```

Comment:
Isn't it too busy now with a 5 buttons row? Screenshot ?
(No alternative idea, and I don't really want a 3rd row, so let's go with that even if busy.)

Reply:
I think it looks ok. It doesn't change the button size much. Honestly the `Add note` button could maybe get renamed to "Note" or something to be a single word for consistency.

![PXL_20231101_002015057](https://github.com/koreader/koreader/assets/904012/34c557ae-3f63-44af-a9a5-2ee25c36149a)

Reply:
(row of 5 + row of 4 looks a bit ugly. Fortunately, people with non-color devices will continue to get quality UI :)

Reply:
I had a vague idea of changing the word "Color" to a highlighter pen icon, but that would probably look weirder.